systemverilog-plugin/UhdmAst.cc

```diff
@@ -2173,13 +2174,64 @@ void UhdmAst::process_typespec_member()
     add_multirange_wire(current_node, packed_ranges, unpacked_ranges);
 }
 
-void UhdmAst::process_enum_typespec()
+void UhdmAst::process_enum_typespec(const UHDM::BaseClass *object)
 {
     // BaseTypespec specifies underlying type of the enum.
     // The BaseTypespec has at most one explicit packed dimension (range).
     // When base type is not specified in SystemVerilog code, it is assumed to be an int.
     // Type of enum items (constants) is the same as the enum type.
     current_node = make_ast_node(AST::AST_ENUM);
+
+    const auto *enum_object = object->Cast<const UHDM::enum_typespec *>();
+    const auto *typespec = enum_object->Base_typespec();
+
+    if (typespec && typespec->UhdmType() == UHDM::uhdmlogic_typespec) {
```

Comment:
We will probably want to reduce a lot of different typespec, so maybe it would be better to use ``switch`` instead of ``if``? WDYT?

Reply:
For now I was going to cover that case of Enums specifically, to cover a case in OpenTitan.

However, it seems to me like the function `reduceExpr` is so generic (it takes `UHDM::any *`) that we could potentially have other typespecs handled the same way, so we could just add more conditions to the `if`.

I wasn't targeting any other typespecs in this PR, so if it's not a problem, I would suggest converting it to `switch` or expanding the `if` statement later with other test cases accordingly.